src/Workspaces/Core/Portable/Classification/ClassificationExtensions.cs

```diff
@@ -4,14 +4,17 @@
 
 #nullable enable
 
+using Microsoft.CodeAnalysis.CSharp;
+
 namespace Microsoft.CodeAnalysis.Classification
 {
     internal static class ClassificationExtensions
     {
         public static string? GetClassification(this ITypeSymbol type)
             => type.TypeKind switch
             {
-                TypeKind.Class => ClassificationTypeNames.ClassName,
+                TypeKind.Class => SymbolDisplayVisitorHelpers.FindValidCloneMethod(type) == null
```

Comment:
my preference on the IDE side wouldb e to have an extension called `.IsRecord(this ITypeSyymbol type)`.  You would impelmeent it by calling into the exposed helper.  However, all IDE code would use this extension which is much more declarative in terms of what it means.